debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -998,6 +991,51 @@ public void readSchema(Tables tables, String databaseCatalog, String schemaNameP
         }
     }
 
+    protected List<ColumnEditor> readTableColumns(ResultSet rs, TableId tableId, ColumnNameFilter columnFilter) throws SQLException {
+        final List<ColumnEditor> cols = new ArrayList<>();
+        final String columnName = rs.getString(4);
+        if (columnFilter == null || columnFilter.matches(tableId.catalog(), tableId.schema(), tableId.table(), columnName)) {
+            ColumnEditor column = Column.editor().name(columnName);
+            column.jdbcType(rs.getInt(5));
+            column.type(rs.getString(6));
+            column.length(rs.getInt(7));
+            if (rs.getObject(9) != null) {
+                column.scale(rs.getInt(9));
+            }
+            column.optional(isNullable(rs.getInt(11)));
+            column.position(rs.getInt(17));
+            column.autoIncremented("YES".equalsIgnoreCase(rs.getString(23)));
+            String autogenerated = null;
+            try {
+                autogenerated = rs.getString(24);
+            }
+            catch (SQLException e) {
+                // ignore, some drivers don't have this index - e.g. Postgres
+            }
+            column.generated("YES".equalsIgnoreCase(autogenerated));
+
+            column.nativeType(resolveNativeType(column.typeName()));
+
+            cols.add(column);
+        }
+        return cols;
+    }
+
+    protected List<String> readPrimaryKeyNames(DatabaseMetaData metadata, TableId id) throws SQLException {
+        List<String> pkColumnNames = null;
+        try (ResultSet rs = metadata.getPrimaryKeys(id.catalog(), id.schema(), id.table())) {
+            while (rs.next()) {
+                if (pkColumnNames == null) {
```

Comment:
Can we make it that the method never returns null, but only an empty list in case there are no PK columns?